dspace-api/src/main/java/org/dspace/health/Core.java

```diff
@@ -0,0 +1,267 @@
+        return sql("SELECT DISTINCT ON (collection_id) collection_id FROM subscription");
+    }
+
+    public static List<TableRow> getHandlesInvalidRows() throws SQLException {
+        List<TableRow> rows = sql("SELECT * FROM handle "
+            + " WHERE NOT ("
+            + "     (handle IS NOT NULL AND resource_type_id IS NOT NULL AND resource_id IS NOT NULL)"
+            + " OR " + "     (handle IS NOT NULL AND url IS NOT NULL)"
+            + " ) ");
+        return rows;
+    }
+
+    // get sizes
+    //
+
+    public static int getItemsTotalCount() throws SQLException {
+        int total = 0;
+        for (java.util.Map.Entry<String, Integer> name_count : getCommunities()) {
+            total += name_count.getValue();
+        }
+        return total;
+    }
+
+    public static int getWorkflowItemsCount() throws SQLException {
+        return sql("SELECT * FROM workflowitem;").size();
+    }
+
+    public static int getNotArchivedItemsCount() throws SQLException {
+        return sql(
+            "SELECT * FROM item WHERE in_archive=false AND withdrawn=false").size();
+    }
+
+    public static int getWithdrawnItemsCount() throws SQLException {
+        return sql("SELECT * FROM item WHERE withdrawn=true").size();
+    }
+
+    public static int getBitstreamsWithoutPolicyCount() throws SQLException {
+        return sql(
+            "SELECT bitstream_id FROM bitstream WHERE deleted<>true AND bitstream_id NOT IN "
+                + "(SELECT resource_id FROM resourcepolicy WHERE resource_type_id=0)")
+            .size();
+    }
+
+    public static int getBitstreamsDeletedCount() throws SQLException {
+        return sql("SELECT * FROM bitstream WHERE deleted=true").size();
+    }
+
+    public static long getHandlesTotalCount() throws SQLException {
+        List<TableRow> rows = sql("SELECT count(1) AS cnt FROM handle");
+        return rows.get(0).getLongColumn("cnt");
+    }
+
+
+    // get more complex information
+    //
+
+    public static List<Map.Entry<String, Integer>> getCommunities()
+        throws SQLException {
+
+        List<Map.Entry<String, Integer>> cl = new java.util.ArrayList<>();
+        Context context = new Context();
+        Community[] top_communities = Community.findAllTop(context);
+        for (Community c : top_communities) {
+            cl.add(
+                new java.util.AbstractMap.SimpleEntry<>(c.getName(), c.countItems())
+            );
+        }
+        context.complete();
+        return cl;
+    }
+
+    public static List<String> getEmptyGroups() throws SQLException {
+        List<String> ret = new ArrayList<>();
+        Context c = new Context();
+        TableRowIterator irows = DatabaseManager
+            .query(c,
+                "SELECT eperson_group_id, "
+                    + "(SELECT text_value FROM metadatavalue "
+                    + "WHERE metadata_field_id=64 AND resource_type_id=6 AND resource_id=eperson_group_id) AS name "
```

Comment:
SELECT eperson_group_id, 
(SELECT text_value FROM metadatavalue 
WHERE metadata_field_id=(SELECT metadata_field_id FROM metadatafieldregistry WHERE element='title' AND qualifier IS NULL AND metadata_schema_id=(SELECT metadata_schema_id FROM metadataschemaregistry WHERE short_id='dc')) AND resource_type_id=6 AND resource_id=eperson_group_id) AS name 
FROM epersongroup 
WHERE eperson_group_id NOT IN (SELECT eperson_group_id FROM epersongroup2eperson)